src/passes/MergeSimilarFunctions.cpp

```diff
@@ -97,6 +97,11 @@
 
 namespace wasm {
 
+// Web limits allow up to 1000 function parameters, but other implementations,
+// such as those on the JVM, only allow up to 255. Do not merge similar
+// functions if the result would require more than this many parameters.
+static constexpr int MaxParams = 255;
```

Comment:
Perhaps add this to `wasm-limits.h` somehow? Outlining etc. might also want to limit the number of params. But can leave as a TODO